docs/userguide/apps.rst

```diff
@@ -89,60 +87,69 @@ Any Parsl app (a Python function decorated with the ``@python_app`` or ``@bash_a
+It is defined by a ``@bash_app`` decorator and the Python code that forms the body of the
+function must return a fragment of Bash shell code to be executed by Parsl.
+The Bash shell code executed by a Bash app can be arbitrarily long. 
+
+The following code snippet presents an example of a Bash app `echo_hello`,
+which returns the bash command `'echo "Hello World!"'` as a string. 
+This string will be executed by Parsl as a Bash command.
 
 .. code-block:: python
 
        @bash_app
        def echo_hello(stderr='std.err', stdout='std.out'):
            return 'echo "Hello World!"'
 
-       # echo_hello() when called will execute the string it returns, creating an std.out file with
-       # the contents "Hello World!"
+       # echo_hello() when called will execute the shell command and
+			 # create a std.out file with the contents "Hello World!"
        echo_hello()
 
-The string thus executed by a Bash app can be arbitrarily long. 
 
 Unlike a Python app, a Bash app cannot return Python objects.
-Instead, it communicates with other functions by creating files.
-A decorated ``@bash_app`` function provides the ``inputs`` and ``outputs`` keyword arguments for managing input and output files.
+Instead, Bash apps communicate with other apps via files.
+A decorated ``@bash_app`` exposes the ``inputs`` and ``outputs`` keyword arguments 
+described above for tracking input and output files.
 It also includes, as described below, keyword arguments for capturing the STDOUT and STDERR streams and recording
 them in files that are managed by Parsl.
 
-
-Limitations
-^^^^^^^^^^^
-
-The following limitations apply to Bash apps:
-
-1. Environment variables are not yet supported.
-
 Special Keywords
 ^^^^^^^^^^^^^^^^
 
-In addition to the ``inputs``, ``outputs``, and ``walltime`` argument keywords described above, a Bash app can take the following keywords:
+In addition to the ``inputs``, ``outputs``, and ``walltime`` keyword arguments
+described above, a Bash app can accept the following keywords:
 
-4. stdout: (string, tuple or `parsl.AUTO_LOGNAME`) The path to a file to which standard output should be redirected. If set to `parsl.AUTO_LOGNAME`, the log will be automatically named according to task id and saved under `task_logs` in the run directory. If set to a tuple `(filename, mode)` then standard output will be redirected to the named file, opened with the specified mode as used by the python `open <https://docs.python.org/3/library/functions.html#open>`_ function.
+4. stdout: (string, tuple or `parsl.AUTO_LOGNAME`) The path to a file to which standard output should be redirected. If set to `parsl.AUTO_LOGNAME`, the log will be automatically named according to task id and saved under `task_logs` in the run directory. If set to a tuple `(filename, mode)` then standard output will be redirected to the named file, opened with the specified mode as used by the Python `open <https://docs.python.org/3/library/functions.html#open>`_ function.
```

Comment:
why do we start with 4 here?  I think this is a bit confusing, given the other list between the 1-3 that this refers to and this list